### ISSUE TITLE
"No signal" connection problem on initial Mac connection attempt

### ISSUE THREAD

**Issue by chrismore** [mozilla-mobile/mozilla-vpn-client#408]
Steps to reproduce:

1) Restart your Mac computer
2) Open the Mozilla VPN Mac client
3) Connect to a server
4) Wait up to 20 seconds

Expected results:

* you remain connected and your exit IP is the location you selected

Actual results:

* You receive a "no signal" message on the app. It sometimes flashes back to connected, but if you check your internet connection, you are not able to load any web pages. If you choose another server and re-connect, it will usually work, but sometimes it takes 3 tries. 

I am only able to replicate this on a reboot. After a successful connection, all is fine. 

I will exam the local log files to see what I can find.

**Comment by ValentinaPC** [comment]
We were able to reproduce following the above steps on Mac OS 10.15.7.

**Comment by ValentinaPC** [comment]
The issue is no longer reproducible, on our side, on the last Mac OS VPN pkg, from 06/01.

**Comment by jess-cook03** [comment]
@lesleyjanenorton @bakulf discussed a solution here:
UI copy/changes would be adding a "Confirming connection..." step between "Connecting..." and "VPN is On". After a connection timeout we could also say something like this in the bi-line: 
`Oops! We ran into an issue...we are trying again...'
Or
'An error occurred while confirming your connection... trying again." 
Or 
'Oops! An error has occurred...we are trying again.'

**Comment by chrismore** [comment]
Thanks, Jess.

@jess-cook03 maybe something for the future. Here is an idea:

What about if we did "Initiating connection..." > "Confirming connection..." > "VPN is on" as the happy path. Also, is communicating connecting vs confirming actually useful to the end user? In theory, it is all in the general bucket of "connecting" regardless what is technically happening on the back end and we could just keep the "Connecting...." state consistent between the connection and confirming state. Just an idea.

**Comment by bakulf** [comment]
For any improvement, let's open a new issue. I'm closing this one.

**Comment by jess-cook03** [comment]
I know we closed this but I just want to put on record the last refinements @lesleyjanenorton and I discussed just a bit ago.

Taking into consideration @chrismore idea above... it made a lot of sense once we saw this live.
We indeed had and will still have the happy path of Connecting > confirming > VPN is on ...... 
We just are surfacing these steps in the UI a bit smoother... meaning the headline copy, if you wish, will stay as 'Connecting..." btw both the connection and confirming phases.... what will change between these phases is the contextual bi-line underneath..... it will start with 'Masking connection and location' then switch to 'Attempting to confirm connection'..... very subtle change that is really only ever seen when there in fact is an issue and its taking longer to connect.

In real time or normal speeds where this all works perfect..... the connecting stage goes so quickly you don't even see the change in strings.

Thank you Chris for that idea.
Thank you @lesleyjanenorton @bakulf so solution-ing it out with me!!! Looks good & makes sense
